server/api/api/endpoints/functions.py

```diff
@@ -718,16 +719,24 @@ def _build_function(
         run_db = server.api.api.utils.get_run_db_instance(db_session)
         fn.set_db_connection(run_db)
 
-        is_nuclio_runtime = fn.kind in RuntimeKinds.nuclio_runtimes()
+        # TODO:  force_build is only used for mlrun build flow, for nuclio runtimes users can specify force_build=True,
+        #  but it is redundant and ignored. Once force_build is removed from RemoteRuntime.deploy() in 1.9.0,
+        #  we can simplify the logic here to:
+        #  is_nuclio_deploy = fn.kind in RuntimeKinds.nuclio_runtimes() and not force_build
+        is_nuclio_deploy = False
+        if fn.kind in RuntimeKinds.pure_nuclio_deployed_runtimes():
+            is_nuclio_deploy = True
+        elif fn.kind == RuntimeKinds.application and not force_build:
```

Comment:
I think it worth leaving the force build in because nuclio does support cache invalidation - we just need to make sure we pass it along